internal/api/api_test.go

```diff
@@ -0,0 +1,164 @@
+	if m.idleError != "" {
+		return fmt.Errorf(m.idleError)
+	}
+	return
+}
+
+func (m *mockOSC) IsIdle(apiURL string, bearerToken string, namespace string, service string) (int, error) {
+	if m.idleError != "" {
+		return -1, fmt.Errorf(m.idleError)
+	}
+	return 1, nil
+}
+
+func (m *mockOSC) WhoAmI(apiURL string, bearerToken string) (string, error) {
+	return "", nil
+}
+
+func (m *mockOSC) WatchBuilds(apiURL string, bearerToken string, buildType string, callback func(model.Object) error) error {
+	return nil
+}
+
+func (m *mockOSC) WatchDeploymentConfigs(apiURL string, bearerToken string, namespaceSuffix string, callback func(model.DCObject) error) error {
+	return nil
+}
+
+type mockClusterView struct{}
+
+func (m *mockClusterView) GetClusters() (r []cluster.Cluster) {
+	return
+}
+
+func (m *mockClusterView) GetDNSView() (r []cluster.DNSView) {
+	return
+}
+
+func (m *mockClusterView) GetToken(openShiftAPIURL string) (string, bool) {
+	return "", true
+}
+
+func (m *mockClusterView) String() string {
+	return ""
+}
+
+type mockResponseWriter struct {
+	buffer       bytes.Buffer
+	writerStatus int
+}
+
+func (m *mockResponseWriter) Header() (h http.Header) {
+	return http.Header{}
+}
+
+func (m *mockResponseWriter) Write(p []byte) (n int, err error) {
+	m.buffer.Write(p)
+	return len(p), nil
+}
+
+func (m *mockResponseWriter) WriteString(s string) (n int, err error) {
+	m.buffer.WriteString(s)
+	return len(s), nil
+}
+
+func (m *mockResponseWriter) WriteHeader(i int) {
+	m.writerStatus = i
+}
+
+func (m *mockResponseWriter) GetBody() string {
+	return m.buffer.String()
+}
+
+type ReqFuncType func(w http.ResponseWriter, r *http.Request, ps httprouter.Params)
+
+func Test_success(t *testing.T) {
+	mockidle := idler{
+		openShiftClient: &mockOSC{},
+		clusterView:     &mockClusterView{},
+	}
+	functions := []ReqFuncType{mockidle.Idle, mockidle.UnIdle, mockidle.IsIdle}
+
+	for _, function := range functions {
+		r, _ := http.NewRequest("GET", "/", nil)
+		q := r.URL.Query()
+		q.Add(OpenShiftAPIParam, "http://localhost")
+		r.URL.RawQuery = q.Encode()
+
+		w := &mockResponseWriter{}
+		function(w, r, nil)
+
+		assert.Equal(t, http.StatusOK, w.writerStatus, fmt.Sprintf("Bad Error Code: %d", w.writerStatus))
+	}
+
+}
+
+func Test_fail(t *testing.T) {
+	mockidle := idler{
+		openShiftClient: &mockOSC{},
+		clusterView:     &mockClusterView{},
+	}
+	functions := []ReqFuncType{mockidle.Idle, mockidle.UnIdle, mockidle.IsIdle}
+	for _, function := range functions {
+		r, _ := http.NewRequest("GET", "/", nil)
+		w := &mockResponseWriter{}
+		function(w, r, nil)
+
+		assert.Equal(t, http.StatusBadRequest, w.writerStatus, fmt.Sprintf("Bad Error Code: %d", w.writerStatus))
+
+		ns := &JSError{}
+		_ = json.Unmarshal(w.buffer.Bytes(), &ns)
+		assert.Equal(t, "OpenShift API URL needs to be specified", ns.Error, fmt.Sprintf("Unexpected error output: %s", ns.Error))
```

Comment:
shouldn't this be `require`?

Reply:
why don't we want to continue to the other tests?

Reply:
If tests are sequential (If first fails, following ones are also likely to fail. Is that the case here?), require is a good idea.

Reply:
I don't think it's that important, but if you feel that strongly I can change that,